actions/inventory.go

```diff
@@ -1,54 +1,136 @@
+
+	// Enable trace logging on the collectors.
+	trace bool
+
+	// a.failOnError when enabled causes ironlib to return an error
+	// if a collector returns an error
+	failOnError bool
+
+	// dynamicCollection when enabled will cause ironlib to
+	// enable collectors based on the detected component vendor.
+	dynamicCollection bool
+
+	// disableCollectorUtilities is the list of collector utilities
+	// to be disabled, this is the name of collector utility
+	// which is returned by its Attributes() method.
+	disableCollectorUtilities []model.CollectorUtility
+}
+
 // Collectors is a struct acting as a registry of various inventory collectors
 type Collectors struct {
-	Inventory          InventoryCollector
-	Drives             []DriveCollector
-	DriveCapabilities  []DriveCapabilityCollector
-	NICs               NICCollector
-	BMC                BMCCollector
-	CPLDs              CPLDCollector
-	BIOS               BIOSCollector
-	TPMs               TPMCollector
-	StorageControllers StorageControllerCollector
+	InventoryCollector
+	NICCollector
+	BMCCollector
+	CPLDCollector
+	BIOSCollector
+	TPMCollector
+	StorageControllerCollector
+	DriveCollectors             []DriveCollector
+	DriveCapabilitiesCollectors []DriveCapabilityCollector
+}
+
+// Option returns a function that sets a InventoryCollectorAction parameter
+type Option func(*InventoryCollectorAction)
+
+// WithTraceLevel sets trace level logging on the action runner.
+func WithTraceLevel() Option {
+	return func(a *InventoryCollectorAction) {
+		a.trace = true
+	}
 }
 
-// InitCollectors constructs the Collectors object
-// with lshw and smartctl collectors by default
+// WithFailOnError sets the InventoryCollectorAction to return on any error
+// that may occur when collecting inventory.
 //
-// param trace causes the collectors to log command outputs
-func InitCollectors(trace bool) *Collectors {
-	return &Collectors{
-		Inventory: utils.NewLshwCmd(trace),
-		Drives: []DriveCollector{
-			utils.NewSmartctlCmd(trace),
-			utils.NewLsblkCmd(trace),
-		},
-		DriveCapabilities: []DriveCapabilityCollector{
-			utils.NewHdparmCmd(trace),
-			utils.NewNvmeCmd(trace),
+// By default the InventoryCollectorAction continues to collect inventory
+// even if one or more collectors fail.
+func WithFailOnError() Option {
+	return func(a *InventoryCollectorAction) {
+		a.failOnError = true
+	}
+}
+
+// DynamicCollection when enabled will cause ironlib to
+// identify collectors based on the detected component vendor.
+func WithDynamicCollection() Option {
+	return func(a *InventoryCollectorAction) {
+		a.dynamicCollection = true
+	}
+}
+
+// WithCollectors sets collectors to the ones passed in as a parameter.
+func WithCollectors(collectors *Collectors) Option {
+	return func(a *InventoryCollectorAction) {
+		a.collectors = *collectors
+	}
+}
+
+// WithDisableCollectorUtilities disables the given collector utilities.
+func WithDisableCollectorUtilities(utilityNames []model.CollectorUtility) Option {
```

Comment:
s/WithDisableCollectorUtilities/WithDisabledCollectorUtilities/g

Reply:
fixed in [2057e09](https://github.com/metal-toolbox/ironlib/pull/75/commits/2057e0930e7282cb218d0230f39567138b082fd3)